From 28b9e4934b3ce248d11b5a60e273876e978c9f70 Mon Sep 17 00:00:00 2001
From: Yixun Lan <dlan@gentoo.org>
Date: Wed, 27 Aug 2025 23:21:09 +0000
Subject: [PATCH 14/33] config: a733: add initial support for Cubie A7A

- Initial support for Radxa Cubie A7A board
- Enable SPL RAM device boot mode, make it able to run U-Boot in DDR
- Enable I2C, PMIC/regulator, pinctrl, SD card, eMMC

Signed-off-by: Yixun Lan <dlan@gentoo.org>
---
 configs/radxa-cubie-a7a_defconfig | 18 ++++++++++++++++++
 1 file changed, 18 insertions(+)
 create mode 100644 configs/radxa-cubie-a7a_defconfig

diff --git a/configs/radxa-cubie-a7a_defconfig b/configs/radxa-cubie-a7a_defconfig
new file mode 100644
index 00000000000..751664db245
--- /dev/null
+++ b/configs/radxa-cubie-a7a_defconfig
@@ -0,0 +1,18 @@
+CONFIG_ARM=y
+CONFIG_ARCH_SUNXI=y
+CONFIG_HAS_CUSTOM_SYS_INIT_SP_ADDR=y
+CONFIG_CUSTOM_SYS_INIT_SP_ADDR=0x4F000000
+CONFIG_DEFAULT_DEVICE_TREE="sun60i-a733-radxa-a7a"
+CONFIG_SPL=y
+CONFIG_MACH_SUN60I_A733=y
+CONFIG_MMC_SUNXI_SLOT_EXTRA=2
+# CONFIG_SYS_MALLOC_CLEAR_ON_INIT is not set
+CONFIG_SPL_LOAD_FIT_ADDRESS=0x4D000000
+CONFIG_BOOTDELAY=3
+CONFIG_SPL_RAM_DEVICE=y
+CONFIG_CMD_I2C=y
+CONFIG_NET_RANDOM_ETHADDR=y
+CONFIG_SYS_I2C_MVTWSI=y
+CONFIG_SUPPORT_EMMC_BOOT=y
+CONFIG_REGULATOR_AXP=y
+# CONFIG_SYSRESET is not set
-- 
2.53.0

